### PR TITLE
Update to Rust v1.51.0

### LINKED ISSUE #252: Re-enable field_reassign_with_default
Similar to cosmwasm [issue 685](https://github.com/CosmWasm/cosmwasm/issues/685).

This depends on a new cosmwasm-schema version, i.e. one that uses schemars 0.8.

### PR DESCRIPTION
**Pull request by maurolacy**
Updates to shiny new Rust v1.51.0.

Closes #252 in passing.

### CODE REVIEW

File: contracts/cw1-subkeys/src/contract.rs [hunk 1]
```diff
@@ -87,9 +87,11 @@ where
     let owner_raw = &deps.api.canonical_address(&info.sender)?;
     // this is the admin behavior (same as cw1-whitelist)
     if cfg.is_admin(owner_raw) {
-        let mut res = Response::default();
-        res.messages = msgs;
-        res.attributes = vec![attr("action", "execute"), attr("owner", info.sender)];
+        let res = Response {
+            messages: msgs,
+            attributes: vec![attr("action", "execute"), attr("owner", info.sender)],
+            ..Response::default()
```
Comment (ethanfrey) [inline contracts/cw1-subkeys/src/contract.rs:93]: Do we want this pattern (okay with me) or explicitly asigning all field?

File: contracts/cw1-whitelist/src/contract.rs [hunk 2]
```diff
@@ -74,9 +74,11 @@ where
     if !can_execute(deps.as_ref(), &info.sender)? {
         Err(ContractError::Unauthorized {})
     } else {
-        let mut res = Response::default();
-        res.messages = msgs;
-        res.attributes = vec![attr("action", "execute")];
+        let res = Response {
+            messages: msgs,
+            attributes: vec![attr("action", "execute")],
+            ..Default::default()
```
Comment (ethanfrey) [inline contracts/cw1-whitelist/src/contract.rs:80]: I know they are the same, but I prefer `Response::default` to `Default::default` when we have access to the explicit types (not just traits)

File: .circleci/config.yml [hunk 3]
```diff
@@ -834,12 +833,12 @@ jobs:
           # Uses --debug for compilation speed
           # FIXME: Change when `check_contract` (part of `cosmwasm-0.14.0`) is published
           #command: cargo install --debug --features iterator --example check_contract -- cosmwasm-vm
-          command: cargo install --debug --features iterator --git https://github.com/CosmWasm/cosmwasm --tag=v0.14.0-beta1 --example check_contract -- cosmwasm-vm
+          command: cargo install --debug --features iterator --git https://github.com/CosmWasm/cosmwasm --branch=main --example check_contract -- cosmwasm-vm
```
Comment (ethanfrey) [inline .circleci/config.yml:836]: I wouldn't make this change here. Rather keep it on a tag (can bump to -beta2 when ready)

File: .circleci/config.yml [hunk 4]
```diff
@@ -792,23 +792,22 @@ jobs:
           command: cargo fmt -- --check
       - run:
           name: Clippy linting on workspace
-          # Silence false positive `field_reassing_with_default` (https://github.com/rust-lang/rust-clippy/issues/6545)
-          command: cargo clippy -- -D warnings -A clippy::field_reassign_with_default
```
Comment (ethanfrey) [inline .circleci/config.yml:796]: :+1:

**Review by ethanfrey — APPROVED** [review]
Looks good

File: contracts/cw1-subkeys/src/contract.rs (hunk 1 above)
Comment (maurolacy) [reply contracts/cw1-subkeys/src/contract.rs:93]: This is the suggested way. I think it's better / more general than the explicit assignment.

File: .circleci/config.yml (hunk 3 above)
Comment (maurolacy) [reply .circleci/config.yml:836]: This should in principle be changed to a released version (i.e. the commented line above). I'm OK with disabling the main version, though.

**Pull request merged**